pkg/cmd/util/clientcmd/factory.go

```diff
@@ -452,6 +468,23 @@ func (f *Factory) Clients() (*client.Client, *kclient.Client, error) {
 	return osClient, kClient, nil
 }
 
+// OriginSwaggerSchema returns a swagger API doc for an Origin schema under the /oapi prefix.
+func (f *Factory) OriginSwaggerSchema(client *kclient.RESTClient, version unversioned.GroupVersion) (*swagger.ApiDeclaration, error) {
+	if version.IsEmpty() {
+		return nil, fmt.Errorf("groupVersion cannot be empty")
+	}
+	body, err := client.Get().AbsPath("/").Suffix("swaggerapi", "oapi", version.Version).Do().Raw()
```

Comment:
Explain calls it only once

On Tue, Feb 9, 2016 at 12:26 AM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/cmd/util/clientcmd/factory.go
> <https://github.com/openshift/origin/pull/7089#discussion_r52269040>:
>
> > @@ -452,6 +468,23 @@ func (f *Factory) Clients() (*client.Client, *kclient.Client, error) {
> >  	return osClient, kClient, nil
> >  }
> >
> > +// OriginSwaggerSchema returns a swagger API doc for an Origin schema under the /oapi prefix.
> > +func (f *Factory) OriginSwaggerSchema(client *kclient.RESTClient, version unversioned.GroupVersion) (*swagger.ApiDeclaration, error) {
> > +	if version.IsEmpty() {
> > +		return nil, fmt.Errorf("groupVersion cannot be empty")
> > +	}
> > +	body, err := client.Get().AbsPath("/").Suffix("swaggerapi", "oapi", version.Version).Do().Raw()
>
> Can this get invoked more than once per process? The swagger doc is huge
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7089/files#r52269040>.
>